src/RuleSet/RuleSets.php

```diff
@@ -67,4 +68,45 @@ public static function getSetDefinition(string $name): RuleSetDescriptionInterfa
 
         return $definitions[$name];
     }
+
+    /**
+     * @param class-string<RuleSetDescriptionInterface> $class
+     */
+    public static function registerRuleSet(string $name, string $class): void
+    {
+        $preg = new Preg();
+
+        if (1 !== $preg->match('/^@[a-z0-9_\/\.-]+$/i', $name)) {
```

Comment:
This is [too optimistic and allows invalid names](https://regex101.com/r/VcOnNr/1). It should be like [here](https://regex101.com/r/VcOnNr/2):

```suggestion
        if (1 !== $preg->match('/^@[a-zA-Z][a-zA-Z0-9_]*(?i)(?<!PhpCsFixer)(?-i)\/[a-zA-Z][a-zA-Z0-9_\-\.]*$/', $name)) {
```

[See as reference](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7064/files#diff-95ff14b92a1d5793b2448a5aadba9352b93197d54d6e2b7431fe9ddefc7a92ebR31) for fixers.

Reply:
PS. if we want to allow vendor or set name starting with `0-9`, then it should be added to first `[]` in both parts. I believe it should not start with non-alphanumeric char, so the rest of chars must be in the second `[]`.

Reply:
I initially thought that no-namespaced ruleset names should be reserved for built-in rulesets, just like it's for fixers. But when I think about my own ruleset it would be better to name it `@Codito` rather than `@Codito/CS` - the `/` would only make naming harder. But on the other hand requiring `vendor/name` with `@PhpCsFixer` allowed only for built-in sets would be a better BC promise - imagine scenario when 3rd party package provides ruleset `@Foo` which is then introduced as built-in... It would break people's configs 🤔.

@kubawerlos @keradus what do you think?

Reply:
It's up to the user to decide how to name the external ruleset when registering it to the config, but of course there can be situations where a ruleset provider provides documentation how to use their ruleset (including naming), and maybe constants to make it look something like this:

```
    $config = new PhpCsFixer\Config();
    return $config
        ->registerCustomRuleSets([
            \MyNameSpace\MyRuleSetClass::NAME => \MyNameSpace\MyRuleSetClass::class,
        ])
        ->setRules([
            \MyNameSpace\MyRuleSetClass::NAME => true,
        ])
    ;
```

I might be wrong, but I would really prefer to have names without namespace. I think the risk of overlapping namespaces is small. This risk could also be part of documentation.